python/PyQt6/core/auto_generated/qgsjsonutils.sip.in

```diff
@@ -290,34 +290,7 @@ Helper utilities for working with JSON and GeoJSON conversions.
 
   public:
 
-    static QgsFeatureList stringToFeatureList( const QString &string, const QgsFields &fields = QgsFields(), QTextCodec *encoding = 0 );
```

Comment:
Instead of removing the whole function, couldn't we just remove the encoding argument and force it to nullptr?

There's the existing SIP_PYARGREMOVE mechanism for this, we could adapt this to a new SIP_PYARGREMOVE6 for selective removal.

Reply:
I would rather migrate the code to new Qt6 API, propose a new QGIS API, and the get rid of SIP_PYQT5_RUN. But later.

Reply:
Hmm - there's definetly plugins out there using this method. See eg https://github.com/koordinates/kart-qgis-plugin/blob/874bc84525e7f173cdf4a2c5cbece08fd6ba2bcc/kart/gui/featurehistorydialog.py#L199 , https://github.com/Raruto/g3w-admin/blob/3b076f382cf8a3ff61d4e8c12df34f937df4a686/g3w-admin/openrouteservice/utils.py#L655

I strongly believe we should leave this function exposed, even if it means that plugins need to adapt and drop the missing QTextCodec argument for qt 6 compatiblity (a LOT of them use it! see https://github.com/search?q=stringToFeatureList+language%3APython&type=code&l=Python, there's many callers specifying a codec).